zerver/lib/reminders.py

```diff
@@ -23,11 +23,12 @@ def get_reminder_formatted_content(message: Message, current_user: UserProfile)
             topic_name=message.topic_name(),
         )
         content = _(
-            "You requested a reminder for the following message sent to [{stream_name} > {topic_name}]({narrow_link})."
+            "You requested a reminder for the following message sent to [{stream_name} > {topic_name}]({narrow_link}/with/{message_id})."
```

Comment:
`zerver/lib/topic_link_util.py` should be the right code for doing the link syntax nicely.